AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -105,53 +105,44 @@ - (void)updateCurrentRangeWithMode:(ASLayoutRangeMode)rangeMode
     _currentRangeMode = rangeMode;
     _didUpdateCurrentRange = YES;
     
-    [self scheduleRangeUpdate];
+    [self performRangeUpdateSynchronously:YES];
   }
 }
 
-- (void)scheduleRangeUpdate
+- (void)performRangeUpdateSynchronously:(BOOL)updateSynchronously
 {
-  if (_queuedRangeUpdate) {
-    return;
+  if (updateSynchronously) {
+    ASDisplayNodeAssertMainThread();
+    [self _updateVisibleNodeIndexPaths];
```

Comment:
In this case, should we set a flag that causes any already-scheduled queued range update to be cancelled?  E.g. something that is checked inside the block below.  Maybe as simple as an if (_queuedRangeUpdate) added inside the block, and setting that flag to NO here.